### PR TITLE
Codeberg support

### PR DESCRIPTION
**Pull request by Jvr2022**
tested

### CODE REVIEW

File: app/src/main/kotlin/com/apkupdater/data/codeberg/CodeBergApp.kt [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    val repo: String,
+    val extra: Regex? = null
+)
+
+val CodeBergApps = listOf(
+    CodeBergApp("eu.kanade.fabsemanga.psyduck", "fabseman", "fabsemanga"),
+    CodeBergApp("com.draco.buoy", "s1m", "savertuner")
+)
+
+fun contains(text: String) = Regex("^.*$text.*\$")
```
Comment (rumboalla) [inline app/src/main/kotlin/com/apkupdater/data/codeberg/CodeBergApp.kt]: contains and notContains are duplicated

File: app/src/main/kotlin/com/apkupdater/repository/CodeBergRepository.kt [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+                    val r = all.flatMap { it }
+                    emit(Result.success(r))
+                }.collect()
+            }
+        }.catch {
+            emit(Result.failure(it))
+            Log.e("CodeBergRepository", "Error searching.", it)
+        }
+
+        private fun selfCheck() = flow {
```
Comment (rumboalla) [inline app/src/main/kotlin/com/apkupdater/repository/CodeBergRepository.kt]: Everything related with selfCheck needs to be removed (it's used for apkupdater updates in GitHub source only)

**Review by rumboalla — CHANGES_REQUESTED** [review]
Looking good, almost there 👍

File: app/src/main/kotlin/com/apkupdater/repository/CodeBergRepository.kt (hunk 2 above)
Comment (Jvr2022) [reply app/src/main/kotlin/com/apkupdater/repository/CodeBergRepository.kt]: fixed

File: app/src/main/kotlin/com/apkupdater/data/codeberg/CodeBergApp.kt (hunk 1 above)
Comment (Jvr2022) [reply app/src/main/kotlin/com/apkupdater/data/codeberg/CodeBergApp.kt]: fixed

**Comment by Jvr2022** [comment]
Ready for review @rumboalla

**Comment by Jvr2022** [comment]
#394

File: app/src/main/kotlin/com/apkupdater/service/CodeBergService.kt [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+import com.apkupdater.data.codeberg.CodeBergRelease
+import retrofit2.http.GET
+import retrofit2.http.Path
+
+interface CodeBergService {
+
+    @GET("/repos/{user}/{repo}/releases")
+    suspend fun getReleases(
+        @Path("user") user: String = "rumboalla",
```
Comment (rumboalla) [inline app/src/main/kotlin/com/apkupdater/service/CodeBergService.kt]: These default values make no sense in here, they were used in GitHub source to find self-updates.

**Review by rumboalla — CHANGES_REQUESTED** [review]

File: app/src/main/kotlin/com/apkupdater/service/CodeBergService.kt (hunk 3 above)
Comment (Jvr2022) [reply app/src/main/kotlin/com/apkupdater/service/CodeBergService.kt]: fixed

**Comment by rumboalla** [comment]
This last commit broke the build

**Comment by Jvr2022** [comment]
I am looking how I can fix it

**Comment by rumboalla** [comment]
Please reopen if you get this working.
As it is:
1) Has conflicts
2) Doesn't build.
3) Even if it built, it would not work. Looks like the CodeBerg API is not implemented properly.
4) Missing basic things. The new repo for example is not added to the Updates/Search viewmodels.

**Pull request closed without merging**